### PR TITLE
Dzelge 52 web preferences

### PR DESCRIPTION
**Pull request by dzelge**
This PR will fix issues in #63. The WebUI will now update all prefernces when:
- Storing the preferences
- The window is closed if the remote/local webapi session is still running
- When a remote session is closed (e.g. the user logs out)

### CODE REVIEW

**Comment by forman** [comment]
@AliceBalfanz here are some suggestions to verify this is working:

### Test 1

- Change panel sizes
- Close tab, reopen tab
- Verify panel sizes are restored

### Test 2

- create new workspace
- in preferences dialog, enable "reopen last workspace"
- Close tab, reopen tab
- Verify last workspace is reopened

### Test 3

- draw some points and polygons
- Close tab, reopen tab
- Verify points and polygons are restored

File: src/renderer/main.tsx [hunk 1]
```diff
@@ -7,10 +7,9 @@ import thunkMiddleware from 'redux-thunk';
 import * as actions from './actions'
 import ApplicationPage from './containers/ApplicationPage'
 import { requireElectron } from './electron';
-import { stateReducer } from './reducers';
 import { State } from './state';
-// import {loadPreferences, updatePreferences} from "./actions";
-
+import { stateReducer } from './reducers';
+import {updatePreferences} from "./actions";
```
Comment (forman) [inline src/renderer/main.tsx:12]:
```suggestion
import { updatePreferences } from "./actions";
```

Please update your WebStorm formatting rules!

File: src/renderer/main.tsx [hunk 2]
```diff
@@ -101,6 +100,15 @@ export function main() {
         event.preventDefault();
         event.stopPropagation();
     });
+
+    window.addEventListener('beforeunload', (event: any) => {
+        event.preventDefault();
+        event.stopPropagation();
```
Comment (forman) [inline src/renderer/main.tsx:106]: Why is stopPropagation required?

**Comment by dzelge** [comment]
closes #63

**Pull request merged**